pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -0,0 +1,325 @@
+		CSIMode:             dk.OneAgent().IsCSIAvailable(),
+	}, nil
+}
+
+func getPaasToken(tokens corev1.Secret) string {
+	if len(tokens.Data[dtclient.PaasToken]) != 0 {
+		return string(tokens.Data[dtclient.PaasToken])
+	}
+
+	return string(tokens.Data[dtclient.ApiToken])
+}
+
+func getAPIToken(tokens corev1.Secret) string {
+	return string(tokens.Data[dtclient.ApiToken])
+}
+
+// getHostMonitoringNodes creates a mapping between all the nodes and the tenantUID for the host-monitoring dynakube on that node.
+// Possible mappings:
+// - mapped: there is a host-monitoring agent on the node, and the dynakube has the tenantUID set => user processes will be grouped to the hosts  (["node.Name"] = "dynakube.tenantUID")
+// - not-mapped: there is NO host-monitoring agent on the node => user processes will show up as individual 'fake' hosts (["node.Name"] = "-")
+// - unknown: there SHOULD be a host-monitoring agent on the node, but dynakube has NO tenantUID set => user processes will restart until this is fixed (node.Name not present in the map)
+//
+// Checks all the dynakubes with host-monitoring against all the nodes (using the nodeSelector), creating the above mentioned mapping.
+func (g *BootstrapperInitGenerator) getHostMonitoringNodes(dk *dynakube.DynaKube) (map[string]string, error) {
+	tenantUUID := dk.Status.OneAgent.ConnectionInfoStatus.TenantUUID
+
+	imNodes := map[string]string{}
+	if !dk.OneAgent().IsCloudNativeFullstackMode() {
+		return imNodes, nil
+	}
+
+	if g.canWatchNodes {
+		var err error
+
+		imNodes, err = g.calculateImNodes(dk, tenantUUID)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		updateImNodes(dk, tenantUUID, imNodes)
+	}
+
+	return imNodes, nil
+}
+
+func (g *BootstrapperInitGenerator) calculateImNodes(dk *dynakube.DynaKube, tenantUUID string) (map[string]string, error) {
+	nodeInf, err := g.initIMNodes()
+	if err != nil {
+		return nil, err
+	}
+
+	nodeSelector := labels.SelectorFromSet(dk.OneAgent().GetNodeSelector(nil))
+	updateNodeInfImNodes(dk, nodeInf, nodeSelector, tenantUUID)
+
+	return nodeInf.imNodes, nil
+}
+
+func updateImNodes(dk *dynakube.DynaKube, tenantUUID string, imNodes map[string]string) {
+	for nodeName := range dk.Status.OneAgent.Instances {
+		if tenantUUID != "" {
+			imNodes[nodeName] = tenantUUID
+		} else if !dk.FeatureIgnoreUnknownState() {
+			delete(imNodes, nodeName)
+		}
+	}
+}
+
+func updateNodeInfImNodes(dk *dynakube.DynaKube, nodeInf nodeInfo, nodeSelector labels.Selector, tenantUUID string) {
+	for _, node := range nodeInf.nodes {
+		nodeLabels := labels.Set(node.Labels)
+		if nodeSelector.Matches(nodeLabels) {
+			if tenantUUID != "" {
+				nodeInf.imNodes[node.Name] = tenantUUID
+			} else if !dk.FeatureIgnoreUnknownState() {
+				delete(nodeInf.imNodes, node.Name)
+			}
+		}
+	}
+}
+
+func (g *BootstrapperInitGenerator) initIMNodes() (nodeInfo, error) {
+	var nodeList corev1.NodeList
+	if err := g.client.List(context.TODO(), &nodeList); err != nil {
```

Comment:
if you follow my recomendation, you can do `g.ctx` here.

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/8d9deb9183a43f3c0402f76ec0be8738de41bb50